data/crac-creation/crac-creator-csa-profiles/src/main/java/com/powsybl/open_rao/data/crac_creation/creator/csa_profile/crac_creator/cnec/GeographicalFilter.java

```diff
@@ -0,0 +1,109 @@
+        Optional<Country> country1 = branch.getTerminal1() == null ? Optional.empty() : getSubstationCountry(branch.getTerminal1().getVoltageLevel().getSubstation());
+        Optional<Country> country2 = branch.getTerminal2() == null ? Optional.empty() : getSubstationCountry(branch.getTerminal2().getVoltageLevel().getSubstation());
+        return country1.equals(country2) ? Set.of(country1) : Set.of(country1, country2);
+    }
+
+    private static Set<Optional<Country>> getSwitchLocation(Switch switchElement) {
+        return Set.of(getSubstationCountry(switchElement.getVoltageLevel().getSubstation()));
+    }
+
+    private static Set<Optional<Country>> getInjectionLocation(Injection<?> injection) {
+        return Set.of(injection.getTerminal() == null ? Optional.empty() : getSubstationCountry(injection.getTerminal().getVoltageLevel().getSubstation()));
+    }
+
+    private static Set<Optional<Country>> getBusLocation(Bus bus) {
+        return Set.of(getSubstationCountry(bus.getVoltageLevel().getSubstation()));
+    }
+
+    private static Set<Optional<Country>> getVoltageLevelLocation(VoltageLevel voltageLevel) {
+        return Set.of(getSubstationCountry(voltageLevel.getSubstation()));
+    }
+
+    private static Set<Optional<Country>> getSubstationLocation(Substation substation) {
+        return Set.of(substation.getCountry());
+    }
+
+    private static Set<Optional<Country>> getHvdcLineLocation(HvdcLine hvdcLine) {
+        return Set.of(
+                hvdcLine.getConverterStation1().getTerminal() == null ? Optional.empty() : getSubstationCountry(hvdcLine.getConverterStation1().getTerminal().getVoltageLevel().getSubstation()),
+                hvdcLine.getConverterStation2().getTerminal() == null ? Optional.empty() : getSubstationCountry(hvdcLine.getConverterStation2().getTerminal().getVoltageLevel().getSubstation())
+        );
+    }
+
+    private static Optional<Country> getSubstationCountry(Optional<Substation> substation) {
+        return substation.isPresent() ? substation.get().getCountry() : Optional.empty();
+    }
+
+    private static Set<Country> filterEmptyCountries(Set<Optional<Country>> countries) {
+        return countries.stream().filter(Optional::isPresent).map(Optional::get).collect(Collectors.toSet());
+    }
+
+    public Set<Country> getNetworkElementsLocations(Set<String> networkElementIds) {
+        Set<Country> locations = new HashSet<>();
+        networkElementIds.forEach(networkElement -> locations.addAll(getNetworkElementLocation(networkElement)));
+        return locations;
+    }
+
+    public boolean networkElementsShareCommonCountry(Set<String> networkElementsSet1, Set<String> networkElementsSet2) {
+        Set<Country> locations1 = getNetworkElementsLocations(networkElementsSet1);
+        Set<Country> locations2 = getNetworkElementsLocations(networkElementsSet2);
+        locations1.retainAll(locations2);
+        return !locations1.isEmpty();
+    }
```

Comment:
make these static by adding Network as argument. This will avoid having to create a GeographicalFilter object for every cnec adder